src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1369,31 +1568,66 @@ private static unsafe int IndexOfAnyProbabilistic(ref char searchSpace, int sear
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int LastIndexOfAny<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> values) where T : IEquatable<T>?
         {
-            if (RuntimeHelpers.IsBitwiseEquatable<T>())
+            if (Unsafe.SizeOf<T>() == sizeof(byte))
```

Comment:
Why did this change?  I don't believe this is valid.  If I use a T that has its own Equals method which provides different semantics than just comparing bit-by-bit, this method will now produce the wrong answer.  That's why IsBitwiseEquatable was being used, to ensure we only vectorize things where we're 100% sure of their notion of equality matches what the vectorized implementations provide.

Reply:
It looks like we are missing a test for this case.

Reply:
it was a bug, I've added the tests and fix it. good catch!